### PR TITLE
Actor streamer type

### PR DESCRIPTION
**Pull request by ziggi**
Issue #137

Any suggestions about code?

And I have some questions about code.
- In SA-MP only 50 actors can be showed for one player, my code completely based on pickup type code and ignores this fact. It must be considered? If yes then how?

For now `globalMaxVisibleItems[STREAMER_TYPE_ACTOR]` is 1000.
- I have add E_STREAMER_HEALTH, E_STREAMER_INVULNERABLE and E_STREAMER_ROTATION to streamer parameter types, it's ok? Or I should be add functions like SetDynamicActorHealth and etc.?

### CODE REVIEW

**Comment by IstuntmanI** [comment]
We should have a "Dynamic" alternative for every native function, it would make the transition easier from SA:MP's functions to streamer's functions.

Missing:
- SetDynamicActor(FacingAngle/Health/Invulnerable/Pos/VirtualWorld)
- GetDynamicActor(FacingAngle/Health/Pos/VirtualWorld)
- IsDynamicActor(Invulnerable/StreamedIn)
- *GetPlayerCameraTargetDynamicActor*
- OnDynamicActorStreamIn/Out

Instead of E_STREAMER_HEALTH, E_STREAMER_INVULNERABLE and E_STREAMER_ROTATION I think you should add set/get functions.

About the player's ability to see only 50 players: they simply won't see any actor above 50, while others could see them ? Well, as long as actors are only global, you would have to check for every player in another loop to see how many they are seeing. I don't think you should make anything for this. Nobody would try to spawn more than 50 actors in the same area, only for testing.

**Comment by ziggi** [comment]
> We should have a "Dynamic" alternative for every native function, it would make the transition easier from SA:MP's functions to streamer's functions.

But for now we do not have similar functions for another types. I want to hear @samp-incognito opinion about this.

**Comment by samp-incognito** [comment]
Nice work with this. I've not reviewed everything, but it all looks pretty good so far.

As far as those other functions are concerned, I agree with @IstuntmanI that it would be ideal. I've tried to mimic existing SA-MP natives/callbacks for every other item type. `GetPlayerCameraTargetDynamicActor` is more than 32 characters, though, so it would need to be called something else.

**Comment by ziggi** [comment]
What about GetPlayerCameraTargetDynActor?

**Comment by samp-incognito** [comment]
That would be okay.

**Comment by samp-incognito** [comment]
I haven't tested any of this, but nothing about the code looks wrong to me, so I'll go ahead and merge it. Thanks for your help, @ziggi.

File: src/natives/actors.cpp [hunk 1]
```diff
@@ -79,3 +79,60 @@ cell AMX_NATIVE_CALL Natives::IsValidDynamicActor(AMX *amx, cell *params)
...
+cell AMX_NATIVE_CALL Natives::GetPlayerTargetDynamicActor(AMX *amx, cell *params)
+{
+	CHECK_PARAMS(1, "GetPlayerTargetDynamicActor");
+	boost::unordered_map<int, Item::SharedActor>::iterator a = core->getData()->actors.find(static_cast<int>(params[1]));
+	if (a != core->getData()->actors.end())
+	{
+		boost::unordered_map<int, int>::iterator i = core->getData()->internalActors.find(a->first);
+		if (i != core->getData()->internalActors.end())
+		{
+			return sampgdk::GetPlayerTargetActor(i->second);
```
Comment (IstuntmanI) [inline src/natives/actors.cpp]: I'm trying to add the missing functions and while I was looking at how other functions are made (well, I'm new in C++) I noticed this function which looks bad to me. GetPlayerTargetActor has a playerid parameter, but this GetPlayerTargetDynamicActor looks like it uses an "actorid" parameter then tries to get that actor's target actor, which isn't possible. So, GetPlayerTargetDynamicActor is ok, or it really has the problem I described ? Sorry if I'm wrong, I'm a noob, for now.

Comment (ziggi) [inline src/natives/actors.cpp]: Yes, this function is wrong, I will fix this soon.

**Pull request merged**